containers/src/Kokkos_DynRankView.hpp

```diff
@@ -477,10 +477,7 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
     return static_cast<int>(m_map.extent(r));
   }
 
-  KOKKOS_INLINE_FUNCTION constexpr typename traits::array_layout layout()
-      const {
-    return m_map.layout();
-  }
+  KOKKOS_INLINE_FUNCTION constexpr typename traits::array_layout layout() const;
```

Comment:
Why did you move the definition out of the class?

Reply:
The new definition depends on the free function `Impl::as_view_of_rank_n` defined below. I don't want to pile on a bunch of forward declarations for the sake of keeping this method definition in-class.